client/data/plugins/installed/use-plugins-query.js

```diff
@@ -0,0 +1,43 @@
+import { useQuery } from 'react-query';
+import { useDispatch } from 'react-redux';
+import wpcom from 'calypso/lib/wp';
+import { receiveSitePlugins, updatePlugin } from 'calypso/state/plugins/installed/actions';
+
+export function getCacheKey( siteIds ) {
+	if ( siteIds.length === 1 ) {
+		return [ 'plugins', siteIds[ 0 ] ];
+	}
+	// The `dev` and `view` query params are not included in the cache key because we want all
+	// the hooks to have the same idea of what the current view is, regardless of dev flags.
+	return [ 'all-plugins' ];
+}
+
+const usePluginsQuery = ( siteIds, queryOptions = {} ) => {
+	let fetch = () => wpcom.req.get( '/me/sites/plugins' );
+	const siteId = siteIds[ 0 ];
+
+	if ( siteIds.length === 1 ) {
+		fetch = () => wpcom.site( siteId ).pluginsList();
+	}
+
+	const dispatch = useDispatch();
+	return useQuery( getCacheKey( siteIds ), fetch, {
+		...queryOptions,
```

Comment:
Seems like it could be a good idea to have `enabled: !! siteIds.length` in order to disable the query if the `siteIds` array is empty.

Reply:
Good catch  this was addressed.

Reply:
This was addressed.